orchestrator/eth_event_watcher.go

```diff
@@ -55,6 +55,39 @@ func (s *peggyOrchestrator) CheckForEvents(
 		return 0, err
 	}
 
+	var erc20DeployedEvents []*wrappers.PeggyERC20DeployedEvent
+	{
+		iter, err := peggyFilterer.FilterERC20DeployedEvent(&bind.FilterOpts{
+			Start: startingBlock,
+			End:   &currentBlock,
+		}, nil)
+		if err != nil {
+			log.WithFields(log.Fields{
+				"start": startingBlock,
+				"end":   currentBlock,
+			}).Errorln("failed to scan past ERC20Deployed events from Ethereum")
+
+			if !isUnknownBlockErr(err) {
+				err = errors.Wrap(err, "failed to scan past ERC20Deployed events from Ethereum")
+				return 0, err
+			} else if iter == nil {
+				return 0, errors.New("no iterator returned")
+			}
+		}
+
+		for iter.Next() {
+			erc20DeployedEvents = append(erc20DeployedEvents, iter.Event)
+		}
+
+		iter.Close()
```

Comment:
I'm not too familiar with the iterator type used here, but should we `defer` closing it in case we hit one of the error cases above?

Reply:
If err != nil, iter will be nil and it will never get opened and the function returns; the only place this can fail is on `iter.Next()`. Also using defer on this case means that the iterator will be kept open until the end of the parent function because the defer belongs to a function, not to a block (had to google this one). So I'm not sure if it helps, or what are the implications, or how to test it properly.

I would like to keep this as-is for now, in the worst-case scenario the GC will take care of it 😅

Reply:
ohh I completely missed that. Nevermind then 👍